servers/src/grin/sync/header_sync.rs

```diff
@@ -143,13 +143,10 @@ impl HeaderSync {
 				if let Some(ref stalling_ts) = self.stalling_ts {
 					if let Some(ref peer) = self.syncing_peer {
 						match self.sync_state.status() {
-							SyncStatus::HeaderSync {
-								current_height: _,
-								highest_height,
-							} => {
+							SyncStatus::HeaderSync { .. } | SyncStatus::BodySync { .. } => {
```

Comment:
Do we even need to check for HeaderSync or BodySync here now?
If we got this far then we know we are syncing headers?

Reply:
In case a HeaderSync state switched to StateSync and spent several minutes on txhashset downloading, and after that, switching back to HeaderSync.
So, perhaps not bad to check the state here.

Reply:
:+1: